lib/puppeteer-helpers.js

```diff
@@ -122,6 +122,35 @@ function getAttribute ($element, attributeName) {
   return $element.evaluate((el, name) => el.getAttribute(name), attributeName)
 }
 
+/**
+ * Snapshots the node in the accessibility tree corresponding to the given element
+ *
+ * @param {import('puppeteer').ElementHandle} $element - Puppeteer element handle
+ * @returns {Promise<import('puppeteer').SerializedAXNode>})} -
+ *   The accessibility tree node corresponding to the element
+ * @throws {Error} If no node corresponds to the element in the accessibility tree
+ *   (for ex. if you query an AT node for a `span` with no role)
+ */
+function getAccessibilityTreeNode ($element) {
+  const node = page.accessibility.snapshot({ root: $element })
+
+  if (!node) {
+    throw new Error('No accessibility tree node exist for this element')
+  }
+
+  return node
+}
+
+/**
+ * Gets the accessible name of the given element, if it exists in the accessibility tree
+ *
+ * @param {import('puppeteer').ElementHandle} $element - Puppeteer element handle
+ * @returns {string} - The element's accessible name
+ */
+async function getAccessibleName ($element) {
+  return (await getAccessibilityTreeNode($element)).name
```

Comment:
Instead of throwing from `getAccessibilityTreeNode` can we do this?

Then the `undefined` name will make it through to the Jest assertion, failing the test

```suggestion
  return (await getAccessibilityTreeNode($element))?.name
```

Reply:
Nah, that'd be like having `getAttribute()` or `getElementById()` crash if there's no attribute or element?

```js
// No accessible name – crashes
expect(await getAccessibleName($something)).toBeUndefined()
```

Not a blocker, make it crash for now if you like

Reply:
Actually there was no need for that throwing 🤦🏻 The code would throw already if trying to access `name` on `undefined` in `getAccessibleName`, so I removed the `getAccessibilityTreeNode` function altogether in favour of a straight call to `puppeteer.accessibility.snapshot()`.